homeassistant/components/ecobee/switch.py

```diff
@@ -0,0 +1,105 @@
+"""Support for using switch with ecobee thermostats."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EcobeeData
+from .const import DOMAIN
+from .entity import EcobeeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+@dataclass(frozen=True, kw_only=True)
+class EcobeeSwitchEntityDescription(SwitchEntity):
+    """Class describing Ecobee switch entities."""
+
+    ecobee_setting_key: str
+    set_fn: Callable[[EcobeeData, int, int], Awaitable]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat switch entity."""
+    data: EcobeeData = hass.data[DOMAIN]
+    _LOGGER.debug("Adding ventilators 20 min switch if present")
+
+    async_add_entities(
+        (
+            EcobeeVentilator20MinSwitch(data, index)
+            for index, thermostat in enumerate(data.ecobee.thermostats)
+            if thermostat["settings"]["ventilatorType"] != "none"
+        ),
+        True,
+    )
+
+
+class EcobeeVentilator20MinSwitch(EcobeeBaseEntity, SwitchEntity):
+    """A Switch class, representing 20 min timer for an ecobee thermostat with ventilator attached."""
+
+    def __init__(
+        self,
+        data: EcobeeData,
+        thermostat_index: int,
+    ) -> None:
+        """Initialize ecobee ventilator platform."""
+        super().__init__(data, thermostat_index)
+        self._attr_unique_id = "ventilator_20m_timer"
```

Comment:
This isn't a unique id. Probably prefix it with the base unique id in the parent.

Reply:
I am still not sure I fully grasp that name vs unique_id concept but I went with:

```python
class EcobeeVentilator20MinSwitch(EcobeeBaseEntity, SwitchEntity):
    """A Switch class, representing 20 min timer for an ecobee thermostat with ventilator attached."""

    _attr_has_entity_name = True
    _attr_name = "Ventilator 20m Timer"

    def __init__(
        self,
        data: EcobeeData,
        thermostat_index: int,
    ) -> None:
        """Initialize ecobee ventilator platform."""
        super().__init__(data, thermostat_index)
        self._attr_unique_id = f"{self.base_unique_id}_ventilator_20m_timer"
```

Is that more inline with other standards, seems to be with what I have seen but not sure

Reply:
The suffix need to be unique within the platform. Ie no other switch entity of this platform should have the same suffix. Right now, there's only the timer switch, but it's good to future proof it.

The thermostat index will make sure that every thermostat gets a unique id, so that two different thermostats don't have the same unique id for their timer switch entities.

Reply:
I don't think we will ever have 2 "ventilator 20 min" switches. Or that would require some re-engineering of their API

It is possible that we have another switch, but if we go this path, we probably need to write another SwitchEntity child class.

Reply:
That combination should be unique. Imthere should be no clash. You can't have 2 of these switches in the physical device